### PR TITLE
Add support for cowboy constraints to Spacesuit

### PR DESCRIPTION
**Pull request by mosche**
@relistan Currently I cannot get cowboy to evaluate the constraints. 
I will push the related spacesuit config in another PR

### CODE REVIEW

**Comment by mosche** [comment]
The corresponding config changes are here:
https://github.com/Nitro/spacesuit-config/pull/36

File: lib/spacesuit/router.ex [hunk 1]
```diff
@@ -40,7 +40,10 @@ defmodule Spacesuit.Router do
       |> process_headers
       |> add_all_actions
 
-    {route, Spacesuit.ProxyHandler, compiled_opts}
+    constraints = Map.get(compiled_opts, :constraints) || []
```
Comment (relistan) [inline lib/spacesuit/router.ex]: I realized there is a form of `Map.get` that takes a default, like `constraints = Map.get(compiled_opts, :constraints, [])`

File: test/spacesuit_router_test.exs [hunk 2]
```diff
@@ -95,12 +95,28 @@ defmodule SpacesuitRouterTest do
...
 
     output = Spacesuit.Router.transform_one_route(route)
-    {_route, _handler, handler_opts} = output
+    {_route, [], _handler, handler_opts} = output
 
     assert [_one, _two] = Map.get(handler_opts, :GET)
     assert [_one, _two] = Map.get(handler_opts, :OPTIONS)
   end
 
+  test "transforming one route with :constraint" do
```
Comment (relistan) [inline test/spacesuit_router_test.exs]: :constraints

**Comment by mosche** [comment]
@relistan Wanna have a quick look on the fix for the validate_routes task?

**Comment by relistan** [comment]
@mosche build is failing on `validate_routes`. I can take a look locally tomorrow.

**Pull request merged**